lib/internal/main/worker_thread.js

```diff
@@ -108,6 +113,19 @@ port.on('message', (message) => {
     publicWorker.parentPort = publicPort;
     publicWorker.workerData = workerData;
 
+    // The counter is only passed to the workers created by the main thread, not
+    // to workers created by other workers.
```

Comment:
… which sounds like something that should at least be tested and/or come with a `TODO` comment to be fixed?